src/test/java/com/wikia/webdriver/elements/mercury/old/SignupPageObject.java

```diff
@@ -61,38 +62,36 @@ public SignupPageObject(WebDriver driver) {
   }
 
   private SignupPageObject typeEmailAddress(String email) {
-    wait.forElementVisible(signupEmail);
-    signupEmail.sendKeys(email);
+    safelySetValue(signupEmail, email);
     return this;
   }
 
   private SignupPageObject typeUsername(String username) {
-    wait.forElementVisible(signupUsername);
-    signupUsername.sendKeys(username);
+    safelySetValue(signupUsername, username);
     return this;
   }
 
   private SignupPageObject typePassword(String password) {
-    wait.forElementVisible(signupPassword);
-    signupPassword.sendKeys(password);
+    safelySetValue(signupPassword, password);
     return this;
   }
 
+  private void safelySetValue(WebElement input, String value) {
+    wait.forElementVisible(input).click();
+    input.clear();
+    input.sendKeys(value);
+    PageObjectLogging.log("Set value",
+      String.format("value of input identified by: %s set to %s",
+        input.getCssValue("class"), value), true);
+  }
+
   private SignupPageObject typeBirthdate(String month, String day, String year) {
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Remove this unused private "typeBirthdate" method. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](http://sonar-s4/coding_rules#rule_key=squid%3AUnusedPrivateMethod)